packages/plugins/typescript-resolvers/tests/mapping.spec.ts

```diff
@@ -0,0 +1,796 @@
+      MyScalar: Scalars['MyScalar'],
+      Int: Scalars['Int'],
+    };`);
+  });
+
+  it('should map to a custom type on every level (+ actual usage in code)', async () => {
+    const testSchema = buildSchema(/* GraphQL */ `
+      type User {
+        id: ID!
+        name: String!
+        chats: [Chat!]
+      }
+
+      type Chat {
+        id: ID!
+        owner: User!
+        members: [User!]
+      }
+
+      type Query {
+        me: User
+      }
+    `);
+    const result = await plugin(
+      testSchema,
+      [],
+      {
+        mappers: {
+          ID: 'number',
+          Chat: 'number',
+        },
+      },
+      { outputFile: '' }
+    );
+
+    const usage = `
+      const resolvers: Resolvers = {
+        Query: {
+          me() {
+            return {
+              id: 1,
+              name: 'Foo',
+              chats: [0,1,2],
+            };
+          }
+        },
+        Chat: {
+          id(parent) {
+            const id: number = parent;
+            return id;
+          }
+        }
+      }
+    `;
+
+    await validate(
+      [result, usage].join('\n\n'),
+      {
+        scalars: {
+          ID: 'number',
+        },
+      },
+      testSchema
+    );
+  });
+
+  it('Should build ResolversTypes with defaultMapper set using {T}', async () => {
+    const result = await plugin(
+      schema,
+      [],
+      {
+        defaultMapper: 'Partial<{T}>',
+      },
+      { outputFile: '' }
+    );
+
+    expect(result).toBeSimilarStringTo(`
+    export type ResolversTypes = {
+      Query: Partial<Query>,
+      MyType: Partial<MyType>,
+      String: Scalars['String'],
+      MyOtherType: Partial<MyOtherType>,
+      Subscription: Partial<Subscription>,
+      Boolean: Scalars['Boolean'],
+      Node: Partial<Node>,
+      ID: Scalars['ID'],
+      SomeNode: Partial<SomeNode>,
+      MyUnion: Partial<MyUnion>,
+      MyScalar: Scalars['MyScalar'],
+      Int: Scalars['Int'],
+    };`);
+  });
+
+  it('Should build ResolversTypes with defaultMapper set using {T} with external identifier', async () => {
+    const result = await plugin(
+      schema,
+      [],
+      {
+        defaultMapper: './my-wrapper#CustomPartial<{T}>',
+      },
+      { outputFile: '' }
+    );
+
+    expect(result).toBeSimilarStringTo(`import { CustomPartial } from './my-wrapper';`);
+    expect(result).toBeSimilarStringTo(`
+    export type ResolversTypes = {
+      Query: CustomPartial<Query>,
+      MyType: CustomPartial<MyType>,
+      String: Scalars['String'],
+      MyOtherType: CustomPartial<MyOtherType>,
+      Subscription: CustomPartial<Subscription>,
+      Boolean: Scalars['Boolean'],
+      Node: CustomPartial<Node>,
+      ID: Scalars['ID'],
+      SomeNode: CustomPartial<SomeNode>,
+      MyUnion: CustomPartial<MyUnion>,
+      MyScalar: Scalars['MyScalar'],
+      Int: Scalars['Int'],
+    };`);
+  });
+
+  it('Should map to a custom type on every level when {T} is used as default mapper', async () => {
```

Comment:
This one throws an error that `number` is not `User`, I think because setting a default mapper (`Partial<{T}>`) makes `ResolversTypes.Chat` to equal `Partial<Chat>` and doesn't not include `Omit` and stuff.

Reply:
You are right. I moved the logic and applied it to any type, even when it's in use with `Partial<{T}>`. 
So now the logic of `Omit` is applied as well (basically, apply it always, expect when there is a specific mapper).